src/nvim/ex_getln.c

```diff
@@ -2349,11 +2349,21 @@ static int command_line_changed(CommandLineState *s)
     //       - Update the screen while the effects are in place.
     //       - Immediately undo the effects.
     State |= CMDPREVIEW;
-    emsg_silent++;  // Block error reporting as the command may be incomplete
-    msg_silent++;   // Block messages, namely ones that prompt
+    emsg_silent++;                 // Block error reporting as the command may be incomplete
+    msg_silent++;                  // Block messages, namely ones that prompt
+    block_autocmds();              // Block events
+    garray_T save_view;
+    win_size_save(&save_view);     // Save current window sizes
+    save_search_patterns();        // Save search patterns
+    State |= PREVIEW_EXEC;         // Set live preview execution state
```

Comment:
does it work if you temporarily _unset_ `CMDPREVIEW` here? or is that insane?

if not, can you at least rename `PREVIEW_EXEC` to `CMDPREVIEW_EXEC`. Use consistent patterns to give readers hints.

Reply:
CMDPREVIEW needs to be set since commands check if Nvim is in cmdpreview mode to know if they should show the preview.

Alright, will rename it